src/librustc_driver/driver.rs

```diff
@@ -1167,15 +1158,11 @@ pub fn phase_5_run_llvm_passes(sess: &Session,
 
 /// Run the linker on any artifacts that resulted from the LLVM run.
 /// This should produce either a finished executable or library.
-#[cfg(feature="llvm")]
-pub fn phase_6_link_output(sess: &Session,
-                           trans: &trans::CrateTranslation,
+pub fn phase_6_link_output<T: TransCrate>(sess: &Session,
+                           trans: &<T as TransCrate>::TranslatedCrate,
```

Comment:
nit: I think you can just use `TRANS::TranslatedCrate`

Reply:
Nope, rustc wants me to disambiguate it.